<!-- -->
<launch>

	<arg name="roadnet_file" default=""/>     <!--默认为空-->
	<arg name="max_speed" default="1" />
	<arg name="use_extern_controller" default="false"/>
	<arg name="which_extern_controller" default="wan_controller"/>
	<arg name="use_car_follower" default="false"/>
	<arg name="use_avoiding" default="false"/>
	
	
	<arg name="rate" default="30" />
	<arg name="delay_time" default="0.3" />
	
	<!-- 底层控制 -->
	<node pkg="pc_stm32" type="pc_stm32" name="pc_stm32">
		<param name="car_state" value="/car_state" />
		<param name="car_goal" value="/car_goal" />
		<param name="pid_params" value="/pid_params" />
		<param name="port_name" value="/dev/stm" />
		<param name="baudrate" value="115200" />
	</node>
	
	<!-- 上层控制 -->
	<node name="driverless_node" pkg="driverless" type="driverless_node" output="screen" launch-prefix="gnome-terminal -x"> <!--  -->
	
		<!-- main -->
		<param name="is_offline_debug" value="false"/>
		<param name="max_speed" value="$(arg max_speed)"/>
		<param name="odom_topic" value="/slam_fused_odom"/>
		
		<!-- path_tracking -->
		<param name="min_foresight_distance" value="4"/>
		<param name="max_side_accel" value="1"/>
		<param name="foreSightDis_speedCoefficient" value = "0.3"/>
		<param name="foreSightDis_latErrCoefficient" value = "1.0"/>
		<param name="car_state" value="/car_state"/>-->
<!--		<param name="omega" value="10.0"/>		-->
<!--		<param name="steer_clearance" value = "0.8"/>  		<!-- 转向间隙 -->-->
<!--		<param name="steer_offset" value = "-1.3"/>     		<!-- 转角补偿 -->-->
<!--		<param name="Ki" value = "0.5"/>			   		<!-- PID-I -->-->
<!--		<param name="tolerate_laterror" value = "0.05"/>	<!-- 容忍横向偏差 -->-->
<!--		<param name="speed" value="$(arg max_speed)"/>-->
<!--		<param name="control_rate" value="$(arg rate)"/>		<!-- Hz -->-->
<!--		<param name="timeout" value="$(arg delay_time)"/>	<!-- 障碍物时延 -->-->
<!--		<param name="car_state" value="/car_state"/>-->
<!--		<param name="car_goal" value="/car_goal"/>-->
<!--		<param name="max_target_yaw_err" value="25.0"/>-->
<!--		<param name="is_object" value="/min_dis_obj"/>-->
<!--		<param name="is_offset" value="/obj_offset"/>-->
<!--		<param name="is_offset_" value="false"/>-->
<!--		<param name="is_back" value="false"/>-->

		<!-- reverse control test-->
		<param name="reverse_test" value="false"/>
		<param name="reverse_path_flip" value="false"/>
		<param name="reverse_path_file" value="$(arg roadnet_file)"/>
	</node>

	<!-- 导远GPS -->
	<include file="$(find daoyuan)/launch/daoyuan.launch"/>
	
	<!-- 镭神雷达 -->
	<include file="$(find lslidar_c16_decoder)/launch/lslidar_c16.launch"/>

	<!-- 大疆雷达 聚类-->
	<include file="$(find livox_ros_driver)/launch/livox_lidar_rviz.launch"/>
	<include file="$(find euclidean_cluster)/launch/euclidean_cluster.launch"/>

</launch>
